bats/tests/containers/volumes.bats

```diff
@@ -0,0 +1,274 @@
+load '../helpers/load'
+
+get_tempdir() {
+    if ! is_windows || ! using_windows_exe; then
+        echo "$BATS_TEST_TMPDIR"
+        return
+    fi
+    # On Windows, create a temporary directory that is in the Windows temporary
+    # directory so that it mounts correctly.  Note that in CI we end up running
+    # with PSModulePath set to pwsh (7.x) paths, and that breaks the code for
+    # PowerShell 5.1.  So we need to have alternative code in that case.
+    # See also https://github.com/PowerShell/PowerShell/issues/14100
+    if command -v pwsh.exe &>/dev/null; then
+        # shellcheck disable=SC2016 # Don't expand PowerShell expansion
+        local command='
+            $([System.IO.Directory]::CreateTempSubdirectory()).FullName
+        '
+        run pwsh.exe -Command "$command"
+        assert_success
+    else
+        # PowerShell 5.1 is built against .net Framework 4.x and doesn't have
+        # [System.IO.Directory]::CreateTempSubdirectory(); create a temporary
+        # file and use its name instead.
+        # shellcheck disable=SC2016 # Don't expand PowerShell expansion
+        local command='
+            $name = New-TemporaryFile
+            Remove-Item -Path $name
+            Start-Sleep -Seconds 1 # Allow anti-virus to do stuff
+            New-Item -Type Directory -Path $name | Out-Null
+            $name.FullName
+        '
+        run powershell.exe -Command "$command"
+        assert_success
+    fi
+    run wslpath -u "$output"
+    assert_success
+    echo "$output" | tr -d "\r"
+}
+
+local_setup() {
+    run get_tempdir
+    assert_success
+    export WORK_PATH=$output
+    run host_path "$WORK_PATH"
+    assert_success
+    export HOST_WORK_PATH=$output
+    export EXPECT_FAILURE=false
+}
+
+local_teardown() {
+    # Only do manual deletion on Windows; elsewhere we use BATS_TEST_TMPDIR so
+    # BATS is expected to do the cleanup.
+    if is_windows && [[ -n $HOST_WORK_PATH ]]; then
+        powershell.exe -Command "Remove-Item -Recurse -LiteralPath '$HOST_WORK_PATH'"
```

Comment:
A few questions, do you think the path needs to be escaped or is it safe to ignore? also, should there be some error handling and logging to know where we fail? or do we get that from `assert_success` when we call it?

I was going to suggest
```
local_teardown() {
    # Only do manual deletion on Windows; elsewhere we use BATS_TEST_TMPDIR so
    # BATS is expected to do the cleanup.
    if is_windows && [[ -n $HOST_WORK_PATH ]]; then
        # Escape the path
        local escaped_path
        escaped_path=$(printf '%q' "$HOST_WORK_PATH")

        if ! powershell.exe -Command "Remove-Item -Recurse -LiteralPath '$escaped_path'" 2>/dev/null; then
            echo "Failed to delete directory: $HOST_WORK_PATH" >&2
            return 1
        fi
        
        echo "Successfully deleted directory: $HOST_WORK_PATH"
    fi
}
```